src/init.cpp

```diff
@@ -57,10 +50,9 @@ template <>
 struct Init<Backend::MC> {
   static void initialize(configuration const& cfg) {
     memory::internal::initializeUmpireHostAllocator(cfg.umpire_host_memory_pool_initial_bytes);
-    // TODO: Consider disabling polling in finalize()
-    if (cfg.mpi_mech == comm::MPIMech::Polling) {
-      pika::mpi::experimental::init(false, cfg.mpi_pool);
-    }
+    // TODO: Is there any need to do polling on the MPI pool? Blocking MPI
+    // communication is also there and keeps polling from happening.
```

Comment:
Can you please explain what you mean exactly?

Reply:
I think I was either confused or just unclear when I wrote this. This is related to the second TODO that I've put in the description ("check if polling actually needs to run on MPI pool"). It's a TODO for the future, that I will leave out of this PR.

In any case, what I mean by this is that polling could run on the default pool (like the CUDA polling) and the MPI pool removed _if_ we make sure that no blocking calls ever happen on the default pool (cf. #512). The MPI pool is not fully utilized at the moment. However, moving the polling to the default pool may also increase latencies or have other bad effects.

Reply:
To do: remove the comment...

Reply:
Comment removed.